File: src/main/java/com/nugurang/graphql/mutation/ProjectMutation.java
package com.nugurang.graphql.mutation;

import com.nugurang.dto.ProjectDto;
import com.nugurang.dto.ProjectInputDto;
import com.nugurang.service.ProjectService;
import graphql.kickstart.tools.GraphQLMutationResolver;
import org.springframework.stereotype.Service;

@Service
public class ProjectMutation implements GraphQLMutationResolver {
    private final ProjectService projectService;

    public ProjectDto createProject(ProjectInputDto projectInputDto, Long team) {
        return projectService.createProject(projectInputDto, team).toDto();
    }

    public ProjectDto updateProject(ProjectInputDto projectInputDto, Long id) {
        return projectService.updateProject(projectInputDto, id).toDto();
    }

    public Long deleteProject(Long id) {
        projectService.deleteProject(id);
        return id;
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ProjectMutation(final ProjectService projectService) {
        this.projectService = projectService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/mutation/ProjectMutation.kt
package com.nugurang.graphql.mutation

import com.nugurang.dto.ProjectDto
import com.nugurang.dto.ProjectInputDto
import com.nugurang.mapper.ProjectMapper
import com.nugurang.service.ProjectService
import graphql.kickstart.tools.GraphQLMutationResolver
import org.springframework.stereotype.Service

@Service
class ProjectMutation(
    private val projectService: ProjectService,
    private val projectMapper: ProjectMapper
) : GraphQLMutationResolver {
    fun createProject(projectInputDto: ProjectInputDto, team: Long): ProjectDto {
        return projectService
            .createProject(projectInputDto, team)
            .let(projectMapper::toDto)
    }

    fun updateProject(projectInputDto: ProjectInputDto, id: Long): ProjectDto {
        return projectService
            .updateProject(projectInputDto, id)
            .let(projectMapper::toDto)
    }

    fun deleteProject(id: Long): Long {
        projectService.deleteProject(id)
        return id
    }
}